registry/storage/manifestlisthandler.go

```diff
@@ -33,16 +34,24 @@ func (ms *manifestListHandler) Unmarshal(ctx context.Context, dgst digest.Digest
 func (ms *manifestListHandler) Put(ctx context.Context, manifestList distribution.Manifest, skipDependencyVerification bool) (digest.Digest, error) {
 	dcontext.GetLogger(ms.ctx).Debug("(*manifestListHandler).Put")
 
-	m, ok := manifestList.(*manifestlist.DeserializedManifestList)
-	if !ok {
+	var schemaVersion int
+	switch m := manifestList.(type) {
+	case *manifestlist.DeserializedManifestList:
+		schemaVersion = m.SchemaVersion
+	case *ocischema.DeserializedImageIndex:
+		schemaVersion = m.SchemaVersion
+	default:
 		return "", fmt.Errorf("wrong type put to manifestListHandler: %T", manifestList)
 	}
+	if schemaVersion != 2 {
```

Comment:
again it would be nice if this were a `const` defined in the individual packages which we could simply import and  use here.